kernel/sched.c

```diff
@@ -1279,6 +1279,9 @@ static inline void z_vrfy_k_thread_priority_set(k_tid_t thread, int prio)
 #ifdef CONFIG_SCHED_DEADLINE
 void z_impl_k_thread_deadline_set(k_tid_t tid, int deadline)
 {
+
+	CLAMP(deadline, 1, INT_MAX);
```

Comment:
```suggestion
	deadline = CLAMP(deadline, 0, deadline);
```

Reply:
Or `MAX(0, deadline)`.  All of those are likely to generate the same code.  The choice of 0 or 1 is essentially arbitrary given the problem domain (any correct use of EDF requires you know ahead of time you can meet the deadline, so both of those would be impossible)

Reply:
Kind of reflecting what Andy already mentioned, I don't see a major gain by clamping... it just (maybe) doesn't do any damage?